### PR TITLE
Added suffix 

### PR DESCRIPTION
**Pull request by RuiAAPeres**

### CODE REVIEW

**Comment by mdiep** [comment]
I'm 👍 to add this, but I think it should be called `concat(value:)`. ☺️ Intra-operator consistency is more important than inter-operator consistency. I could also imagine a `prefix` variant that takes a `SignalProducer`.

**Comment by RuiAAPeres** [comment]
@mdiep what do you think about this:

```swift
	@warn_unused_result(message="Did you forget to call `start` on the producer?")
	public func concat(next: SignalProducer<Value, Error>) -> SignalProducer<Value, Error> {
		return SignalProducer<SignalProducer<Value, Error>, Error>(values: [ self.producer, next ]).flatten(.Concat)
	}
	
	/// `concat`s `value` onto `self`.
	@warn_unused_result(message="Did you forget to call `start` on the producer?")
	public func concat(value value: Value) -> SignalProducer<Value, Error> {
		return self.concat(SignalProducer(value: value))
	}
	
	/// `concat`s `self` onto initial `previous`.
	@warn_unused_result(message="Did you forget to call `start` on the producer?")
	public func prefix(previous previous: SignalProducer<Value, Error>) -> SignalProducer<Value, Error> {
		return previous.concat(self.producer)
	}
	
	/// `concat`s `self` onto initial `value`.
	@warn_unused_result(message="Did you forget to call `start` on the producer?")
	public func prefix(value value: Value) -> SignalProducer<Value, Error> {
		return self.prefix(previous: SignalProducer(value: value))
	}
```

More specifically the `previous`.

**Comment by mdiep** [comment]
I think the `SignalProducer` variants should use unnamed parameters. I think of those as the default parameter type. The argument name is used when the parameter is something other than a `SignalProducer`/`Signal`.

**Comment by RuiAAPeres** [comment]
> The argument name is used when the parameter is something other than a SignalProducer/Signal

👍 

And as for the `previous` naming?

**Comment by mdiep** [comment]
> And as for the previous naming?

That seems good. 👍

File: ReactiveCocoa/Swift/Flatten.swift [hunk 1]
```diff
@@ -385,11 +385,23 @@ extension SignalProducerType {
...
+	
+	/// `concat`s `value` onto `self`.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func concat(value value: Value) -> SignalProducer<Value, Error> {
+		return self.concat(SignalProducer(value: value))
+	}
+	
+	/// `concat`s `self` onto initial `previous`.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func prefix(previous: SignalProducer<Value, Error>) -> SignalProducer<Value, Error> {
```
Comment (ikesyo) [inline ReactiveCocoa/Swift/Flatten.swift]: I'd like to accept a `SignalProducerType` here:

```swift
public func prefix<S: SignalProducerType where P.Value == Value, P.Error == Error>(previous: P) -> SignalProducer<Value, Error> {
```

**Comment by RuiAAPeres** [comment]
@ikesyo 👍

**Comment by mdiep** [comment]
This is great—thanks! And just in time for the 4.2 release. ☺️

**Pull request merged**